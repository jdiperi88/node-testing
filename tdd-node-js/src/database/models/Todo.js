const Sequelize = require("sequelize");
const dbName = "todos_db";

const db = new Sequelize({
  database: dbName,
  dialect: "postgres",
  define: {
    underscored: true
  }
});

const Todo = db.define("todo", {
  id: {
    type: Sequelize.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  title: Sequelize.STRING,
  description: Sequelize.STRING,
  completed: Sequelize.BOOLEAN
});

db.sync();

module.exports = {
  Todo
};
