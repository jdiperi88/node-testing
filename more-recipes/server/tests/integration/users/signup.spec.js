import supertest from "supertest";
import app from "../../../index";
import { User } from "../../../database/models";

describe("user sign up test", () => {
  test("should register a new user", async () => {
    //arrange
    //get some fake user data
    const fakeUser = {
      name: "joey dip",
      email: "jdip@gmail.com",
      password: "password"
    };
    //action
    const response = await supertest(app)
      .post("/api/v1/users/signup")
      .send(fakeUser);
    //make post request

    //assertion
    //1. the response has the user data
    expect(response.status).toBe(200);
    expect(response.body.data.user.email).toBe(fakeUser.email);
    expect(response.body.data.user.name).toBe(fakeUser.name);
    expect(response.body.data.access_token).toBeTruthy();
    //2. the database has a user with the credentials we signed up with.
    const userFromDatabase = await User.find({
      where: { email: fakeUser.email }
    });

    expect(userFromDatabase).toBeTruthy();
    User.destroy({ where: {} });
  });

  test("should return validation error for duplicate email", async () => {
    //arrange
    const fakeUser = {
      name: "joey dip",
      email: "jdip@gmail.com",
      password: "password"
    };
    //action
    //make post request
    await User.create(fakeUser);
    const response = await supertest(app)
      .post("/api/v1/users/signup")
      .send(fakeUser);

    expect(response.status).toBe(422);

    expect(response.body.data).toMatchSnapshot();
    User.destroy({ where: {} });
  });
});
