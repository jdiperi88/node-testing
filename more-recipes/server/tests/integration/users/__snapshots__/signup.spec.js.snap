// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user sign up test should return validation error for duplicate email 1`] = `
Object {
  "errors": Array [
    "A user with this email already exists.",
  ],
}
`;
